lib/compress/zstd_compress_internal.h

```diff
@@ -600,14 +577,30 @@ static void ZSTD_safecopyLiterals(BYTE* op, BYTE const* ip, BYTE const* const ie
     while (ip < iend) *op++ = *ip++;
 }
 
+#define ZSTD_REP_MOVE     (ZSTD_REP_NUM-1)
+#define STORE_REPCODE_1 STORE_REPCODE(1)
+#define STORE_REPCODE_2 STORE_REPCODE(2)
+#define STORE_REPCODE_3 STORE_REPCODE(3)
+#define STORE_REPCODE(r) (assert((r)>=1), assert((r)<=3), (r)-1)
+#define STORE_OFFSET(o)  (assert((o)>0), o + ZSTD_REP_MOVE)
```

Comment:
Note that this pattern means evaluating the argument must not have any side effects.

Reply:
Yes, this is a known side effect of macros.
Indeed, these macros shall only be used with scalar variables, not invoking function calls.

A possible alternative would be to swap these macros with `inline` functions,
although this second solution introduces its own set of potential casting issues.